frontend/src/metabase/search/components/SearchSidebar/SearchSidebar.tsx

```diff
@@ -1,59 +1,104 @@
 import _ from "underscore";
+import { useCallback, useMemo } from "react";
 import type {
   FilterTypeKeys,
+  SearchFilterComponent,
   SearchFilterPropTypes,
   URLSearchFilterQueryParams,
-  SearchSidebarFilterComponent,
 } from "metabase/search/types";
 import { Stack } from "metabase/ui";
 import { SearchFilterKeys } from "metabase/search/constants";
-import { SidebarFilter } from "metabase/search/components/SidebarFilter/SidebarFilter";
+import { DropdownSidebarFilter } from "metabase/search/components/SearchSidebar/DropdownSidebarFilter/DropdownSidebarFilter";
 import { TypeFilter } from "metabase/search/components/filters/TypeFilter/TypeFilter";
+import { PLUGIN_CONTENT_VERIFICATION } from "metabase/plugins";
+import { ToggleSidebarFilter } from "metabase/search/components/SearchSidebar/ToggleSidebarFilter/ToggleSidebarFilter";
 import { CreatedByFilter } from "metabase/search/components/filters/CreatedByFilter/CreatedByFilter";
 
 type SearchSidebarProps = {
   value: URLSearchFilterQueryParams;
   onChange: (value: URLSearchFilterQueryParams) => void;
 };
 
-export const filterMap: Record<FilterTypeKeys, SearchSidebarFilterComponent> = {
-  [SearchFilterKeys.Type]: TypeFilter,
-  [SearchFilterKeys.CreatedBy]: CreatedByFilter,
-};
-
 export const SearchSidebar = ({ value, onChange }: SearchSidebarProps) => {
+  const filterMap: Record<FilterTypeKeys, SearchFilterComponent | null> =
+    useMemo(
+      () => ({
+        [SearchFilterKeys.Type]: TypeFilter,
+        [SearchFilterKeys.CreatedBy]: CreatedByFilter,
+        [SearchFilterKeys.Verified]: PLUGIN_CONTENT_VERIFICATION.VerifiedFilter,
+      }),
+      [],
+    );
+
+  const isValidFilterValue = useCallback(
+    (
+      key: FilterTypeKeys,
+      val: SearchFilterPropTypes[FilterTypeKeys],
+    ): boolean => {
+      if (!val || !filterMap[key]) {
+        return false;
+      }
+
+      if (Array.isArray(val)) {
+        return val.length > 0;
+      }
+      return true;
+    },
+    [filterMap],
+  );
+
   const onOutputChange = (
     key: FilterTypeKeys,
     val: SearchFilterPropTypes[FilterTypeKeys],
   ) => {
-    if (!val || (Array.isArray(val) && val.length === 0)) {
+    if (!isValidFilterValue(key, val)) {
       onChange(_.omit(value, key));
     } else {
-      const { toUrl } = filterMap[key];
+      const filterMapElement = filterMap[key];
+      const toUrl = filterMapElement?.toUrl;
       onChange({
         ...value,
-        [key]: toUrl(val),
+        [key]: toUrl?.(val) ?? val,
       });
     }
   };
 
   const getFilter = (key: FilterTypeKeys) => {
     const Filter = filterMap[key];
-    const filterValue = Filter.fromUrl(value[key]);
-    return (
-      <SidebarFilter
-        filter={Filter}
-        data-testid={`${key}-search-filter`}
-        value={filterValue}
-        onChange={value => onOutputChange(key, value)}
-      />
-    );
+
+    if (!Filter) {
+      return null;
+    }
```

Comment:
one way to simplify this is to have a default value of the component from enterprise folder being a component that renders nothing instead of null e.g.
```
() => null
```

This will ensure we don't need a special code to deal with it and it should simplify its relevant parts quite a bit too.